### ISSUE TITLE
The visualization of  image warping

### ISSUE THREAD

**Issue by EDENpraseHAZARD** [ClementPinard/SfmLearner-Pytorch#102]
Hi Clement, recently I used the your pretrained model `dispnet_model_best.pth.tar` and `exp_pose_model_best.pth.tar` to warp the images, but got bad result.
I used 5-frame snippets as input, and show the target view frame_2 and its nearby view frame_1 here :
target view:  
![img_2](https://user-images.githubusercontent.com/53077461/87127700-ea61aa80-c2c0-11ea-8541-e6c3a9ada6e0.png)
nearby view:
![img_1](https://user-images.githubusercontent.com/53077461/87127727-fa798a00-c2c0-11ea-94c7-9eb981f8f792.png) 
Than I got the warped view :
![inverse_warped_1](https://user-images.githubusercontent.com/53077461/87127817-2eed4600-c2c1-11ea-8af1-d08a0131c1c4.png)

I don't care about distortion which suffered from semi_dense depth prediction since I have the depth ground truth. However, the pose is also inaccurate and the displacement showed between the warped view and the nearby view is too small .I have read the issues of this project and find that the pose groundtruth provided by KITTI is inaccurate .
What's more, I notice that the best pose result you showed on the README is test from `exp_pose_checkpoint.pth.tar` rather than `exp_pose_model_best.pth.tar`. So I also used `exp_pose_checkpoint.pth.tar` and `dispnet_model_best.pth.tar` to do the warping and got worse result:
![inverse_warped_1](https://user-images.githubusercontent.com/53077461/87128638-8dff8a80-c2c2-11ea-896c-a226034bd0ae.png)
I don't know the reason and think that it may caused by the dismatch of  `exp_pose_checkpoint.pth.tar` and `dispnet_model_best.pth.tar` for I don't have `dispnet_checkpoint.pth.tar`.

Moreover, I haved tried many experiments of image warping used SfMLearner but the results were always disappointed. So Clement, have you ever tried image warping and got some good result ? I guess the pose predicted by the posenet is not accurate enough to do image warping.

**Comment by ClementPinard** [comment]
Hi, I highly suspect the missing normalization. Depth and pose models have not been trained together in the pretrained network folder, and thus warping is not expected to be perfect, but at least it should show some structure, contrary to this LSD soup.

The network expects pictures with pixels in the range [-1, 1].

If you correctly applied the normalization, you might want to see if a network that you trained yourself, even with worse numerical test results than the pretrained one performs better or not, which would indicate some bug in your code.

At a last measure, I'll try to look up what's wrong with my pretrained network.

Clément

**Comment by EDENpraseHAZARD** [comment]
I check my test code which really do the normalization and the pixels are in the range [-1, 1]
Here is the test code:
[test_warp.zip](https://github.com/ClementPinard/SfmLearner-Pytorch/files/4916056/test_warp.zip)
Moreover I tried to used the model I finetuned with Cityscapes to do image warping, and here is the result :
target image:
![img_2](https://user-images.githubusercontent.com/53077461/87371138-4d03c080-c5b7-11ea-8730-6ff246cd1a21.png)
nearby image:
![img_0](https://user-images.githubusercontent.com/53077461/87371148-53923800-c5b7-11ea-828f-c0fca4162f51.png)
warped image:
![inverse_warped_0](https://user-images.githubusercontent.com/53077461/87371160-5a20af80-c5b7-11ea-997b-8a6c6c8dab5f.png)
Well you can see, the warped image has a little distortion and shows good structure. However, the wapred image is in alignment with nearby image but not target image which means the pred_pose is inaccurate and the displacement between nearby image and warped image is little.
So I guess the pose predicted from posenet is not accurate enough to do image warping

**Comment by huhaoyue** [comment]
> I check my test code which really do the normalization and the pixels are in the range [-1, 1] Here is the test code: [test_warp.zip](https://github.com/ClementPinard/SfmLearner-Pytorch/files/4916056/test_warp.zip) Moreover I tried to used the model I finetuned with Cityscapes to do image warping, and here is the result : target image: ![img_2](https://user-images.githubusercontent.com/53077461/87371138-4d03c080-c5b7-11ea-8730-6ff246cd1a21.png) nearby image: ![img_0](https://user-images.githubusercontent.com/53077461/87371148-53923800-c5b7-11ea-828f-c0fca4162f51.png) warped image: ![inverse_warped_0](https://user-images.githubusercontent.com/53077461/87371160-5a20af80-c5b7-11ea-997b-8a6c6c8dab5f.png) Well you can see, the warped image has a little distortion and shows good structure. However, the wapred image is in alignment with nearby image but not target image which means the pred_pose is inaccurate and the displacement between nearby image and warped image is little. So I guess the pose predicted from posenet is not accurate enough to do image warping

Hi, I download your test_warp.zip, but it doesn't have the relevant work. Hope you can provide another copy, thank you very much！(786059151@qq.com)
